cmd/flux/create_source_git.go

```diff
@@ -236,6 +244,15 @@ func createSourceGitCmdRun(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	if provider := sourceGitArgs.provider.String(); provider != "" {
+		switch provider {
+		case sourcev1.GitProviderAzure:
+			gitRepository.Spec.Provider = sourcev1.GitProviderAzure
+		case sourcev1.GitProviderGeneric:
+			gitRepository.Spec.Provider = sourcev1.GitProviderGeneric
+		}
+	}
```

Comment:
Sure, removing the source API reference, however since `.spec.provider` is an optional field, continuing to set it outside the object if `--provider` value is not empty.